monitoring/ping.go

```diff
@@ -183,29 +188,38 @@ func (c *pingChecker) checkNodesRTT(nodes []serf.Member, client *serf.RPCClient)
 }
 
 // storePingInHDR is used to store ping RoundTrip values in HDR Histograms in memory
-func (c *pingChecker) storePingInHDR(pingRttStats int64, node serf.Member) error {
-	nodeTTLMapInterface, exists := c.rttStats.Get(node.Name)
+func (c *pingChecker) storePingInHDR(pingroundtripLatency int64, node serf.Member) error {
+	nodeTTLMapInterface, exists := c.roundtripLatency.Get(node.Name)
 	if !exists {
-		c.rttStats.Set(node.Name,
+		c.roundtripLatency.Set(node.Name,
 			hdrhistogram.New(pingRoundtripMinimum.Nanoseconds(),
 				pingRoundtripMaximum.Nanoseconds(),
 				pingRoundtripSignificativeFigures),
 			statsTTLPeriod)
-		nodeTTLMapInterface, _ = c.rttStats.Get(node.Name)
+		nodeTTLMapInterface, _ = c.roundtripLatency.Get(node.Name)
 	}
 
-	nodeTTLMap := nodeTTLMapInterface.(*hdrhistogram.Histogram)
+	nodeTTLMap, ok := nodeTTLMapInterface.(*hdrhistogram.Histogram)
+	if !ok {
+		errMsg := fmt.Sprintf("couldn't parse roundtripLatency as HDRHistogram on %s",
+			c.serfMemberName)
+		return errors.New(errMsg)
```

Comment:
please use `trace.BadParameter("... %v", ...)` here instead of errors package here and everywhere else